realm/realm-library/src/main/cpp/CMakeLists.txt

```
@@ -130,12 +131,43 @@ create_javah(TARGET jni_headers
     DEPENDS ${classes_PATH}
 )
 
-option(REALM_JAVA_BUILD_CORE_FROM_SOURCE "Build Realm Core from source, as opposed to downloading prebuilt binaries" ON)
-if(REALM_JAVA_BUILD_CORE_FROM_SOURCE)
+# Check if user defined download location
+if(NOT DEFINED ENV{REALM_CORE_DOWNLOAD_DIR})
+    set(REALM_CORE_DOWNLOAD_DIR ${CMAKE_BINARY_DIR})
+else()
+    set(REALM_CORE_DOWNLOAD_DIR $ENV{REALM_CORE_DOWNLOAD_DIR})
+endif()
+
+if(REALM_JAVA_BUILD_CORE_FROM_SOURCE OR NOT REALM_ENABLE_SYNC)
+    message(STATUS "Building Realm Core from source...")
+
     set(REALM_BUILD_LIB_ONLY ON)
     add_subdirectory(realm-core)
```

```suggestion
    add_subdirectory(realm-core EXCLUDE_FROM_ALL)
```
This is something I should have suggested sooner - this option causes any targets defined in `realm-core` to not be included in the default `ALL` target generated by CMake. Only targets explicitly referenced (like `Realm::Storage`, `Realm::Sync`, and `Realm::ObjectStore`) will be built.